src/webui/webapplication.cpp

```diff
@@ -306,9 +308,31 @@ void WebApplication::action_command_shutdown()
 void WebApplication::action_command_download()
 {
     CHECK_URI(0);
-    CHECK_PARAMETERS("urls");
+    CHECK_PARAMETERS("urls" << "savepath" << "label" << "cookie");
```

Comment:
That won't work.  CHECK_PARAMETERS checks that exactly those parameters are in the request.  No more, no less.  I would have to delete the line completely, which is fine, and several other functions don't have it.